libraries/botbuilder/tests/teamsActivityHandler.test.js

```diff
@@ -508,6 +508,38 @@ describe('TeamsActivityHandler', () => {
                 .catch(err => done(err));
         });
 
+        it('handleTeamsTabFetch is not overridden', async () => {
+            const bot = new TeamsActivityHandler();
+
+            const adapter = new TestAdapter(async (context) => {
+                await bot.run(context);
+            });
+
+            const taskFetchActivity = createInvokeActivity('tab/fetch');
+            await adapter.send(taskFetchActivity)
+                .assertReply((activity) => {
+                    assert.strictEqual(activity.type, 'invokeResponse');

```

Comment:
One last test suggestion: use `assert.deepStrictEqual`. This could simply be:

```typescript
assert.deepStrictEqual(activity, {
  type: 'invokeResponse',
  value: {
    status: 501,
    body: undefined,
  },
});
```

Among other things, it prints a better diff for failure scenarios.

Reply:
Because `activity` has more keys than just `type` and `value`, I had to adjust this a little, but still streamlined it a tad.

Reply:
I _think_ the code is smart enough to only assert key/values you express in the `expected` arg.

Reply:
Unfortunately, it was not. It also didn't like having `body: undefined` in there since it's omitted entirely in the activity.

Reply:
Fair enough - not worth spending too many cycles on anyways!